src/terms/kinetic.jl

```diff
@@ -12,8 +12,9 @@ struct TermKinetic <: Term
     kinetic_energies::Vector{Vector} # kinetic energy 1/2|G+k|^2 for every kpoint
 end
 function TermKinetic(basis::PlaneWaveBasis; scaling_factor=1)
-    kinetic_energies = [[scaling_factor * sum(abs2, G + kpt.coordinate_cart) / 2
-                         for G in G_vectors_cart(kpt)]
+    T = eltype(basis)
+    factor = scaling_factor / T(2)
```

Comment:
I don't really like this, it doesn't make logical sense: the 1/2 and the scaling_factor belong to different parts of the formula, it doesn't make much sense to group them. Also is T(2) really necessary?

Reply:
T(2) Yes, otherwise you get promotion to Float64 by default.